pkg/ddl/sequence.go

```diff
@@ -45,41 +45,40 @@ func onCreateSequence(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ err
 		return ver, errors.Trace(err)
 	}
 
+	err = createSequenceWithCheck(t, job, tbInfo)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
 	ver, err = updateSchemaVersion(d, t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
+	job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tbInfo)
+	return ver, nil
+}
 
+func createSequenceWithCheck(t *meta.Meta, job *model.Job, tbInfo *model.TableInfo) error {
 	switch tbInfo.State {
-	case model.StateNone:
+	case model.StateNone, model.StatePublic:
```

Comment:
Why add model.StatePublic here?

Reply:
The reason is the same as https://github.com/pingcap/tidb/pull/51870. Maybe we can set this state on caller side, let's refactor these in next PR.